src/test/extensions/changelog.spec.ts

```diff
@@ -0,0 +1,112 @@
+import * as chai from "chai";
+import { expect } from "chai";
+chai.use(require("chai-as-promised"));
+import * as sinon from "sinon";
+
+import * as changelog from "../../extensions/changelog";
+import * as extensionApi from "../../extensions/extensionsApi";
+
+function testExtensionVersion(
+  version: string,
+  releaseNotes?: string
+): extensionApi.ExtensionVersion {
+  return {
+    name: `publishers/test/extensions/test/versions/${version}`,
+    ref: `test/test@${version}`,
+    hash: "abc123",
+    sourceDownloadUri: "https://google.com",
+    releaseNotes,
+    spec: {
+      name: "test",
+      version,
+      resources: [],
+      params: [],
+      sourceUrl: "https://google.com",
+    },
+  };
+}
+
+describe("changelog", () => {
+  describe("GetReleaseNotesForUpdate", () => {
+    let listExtensionVersionStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      listExtensionVersionStub = sinon.stub(extensionApi, "listExtensionVersions");
+    });
+
+    afterEach(() => {
+      listExtensionVersionStub.restore();
+    });
+
+    it("should return release notes for each version in the update", async () => {
+      const extensionVersions: extensionApi.ExtensionVersion[] = [
+        testExtensionVersion("0.1.1", "foo"),
+        testExtensionVersion("0.1.2", "bar"),
+      ];
+      listExtensionVersionStub
+        .withArgs("test/test", `id<="0.1.2" AND id>"0.1.0"`)
+        .returns(extensionVersions);
+      const want = {
+        "0.1.1": "foo",
+        "0.1.2": "bar",
+      };
+
+      const got = await changelog.getReleaseNotesForUpdate("test/test", "0.1.0", "0.1.2");
+
+      expect(got).to.deep.equal(want);
+    });
+
+    it("should exclude versions that don't have releaseNotes", async () => {
```

Comment:
Or should we say no release note? providers forgetting to write release note might be pretty common

Reply:
Gonna leave this as is since:
- We're gonna check for a CHANGELOG entry and enforce that it should be there on ext:publish, so this should not happen frequently
- In the case where an extension doesn't have release notes populated on the backend, it may still have release notes available elsewhere, so it could be misleading.

Reply:
sgtm, thanks!